src/pal/src/map/map.cpp

```diff
@@ -2303,27 +2456,33 @@ void * MAPMapPEFile(HANDLE hFile)
         || (preferredBase + virtualSize < preferredBase)    // Does the image overflow?
         )
     {
-        ERROR_(LOADER)( "image is corrupt\n" );
+        if (!doPreload)
+        {
+            ERROR_(LOADER)( "image is corrupt\n" );
```

Comment:
What do you think about possibility to move this into a helper macro, that would take two arguments - the message and the condition?